sdk/search/search-documents/src/generated/service/models/index.ts

```diff
@@ -541,6 +543,38 @@ export interface CognitiveServicesAccount {
   description?: string;
 }
 
+/** Definition of additional projections to azure blob, table, or files, of enriched data. */
+export interface SearchIndexerKnowledgeStore {
+  /** The connection string to the storage account projections will be stored in. */
+  storageConnectionString: string;
+  /** A list of additional projections to perform during indexing. */
+  projections: SearchIndexerKnowledgeStoreProjection[];
+}
+
+/** Container object for various projection selectors. */
+export interface SearchIndexerKnowledgeStoreProjection {
+  /** Projections to Azure Table storage. */
+  tables?: SearchIndexerKnowledgeStoreTableProjectionSelector[];
+  /** Projections to Azure Blob storage. */
+  objects?: SearchIndexerKnowledgeStoreObjectProjectionSelector[];
+  /** Projections to Azure File storage. */
+  files?: SearchIndexerKnowledgeStoreFileProjectionSelector[];
```

Comment:
Are they properties actually optional? The docs lead me to believe they are always returned, though sometimes the arrays are empty: https://docs.microsoft.com/en-us/azure/search/knowledge-store-projection-overview#defining-a-table-projection